analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/InvalidCastToInterface.CSharp8.cs

```diff
@@ -0,0 +1,114 @@
+﻿public class NullCoalescenceAssignment
+{
+    public void NullCoalescenceAssignment_Null()
+    {
+        int? i1 = null;
+        i1 ??= (int)i1; // FIXME Non-compliant {{Nullable is known to be empty, this cast throws an exception.}}
+    }
+
+    public void NullCoalescenceAssignment_NotNull()
+    {
+        int? i1 = 1;
+        i1 ??= (int)i1;
+    }
+
+    public void NullCoalescenceAssignmentResult_Null()
+    {
+        int? i = null;
+        i ??= null;
+        var r1 = (int)i; // FIXME Non-compliant
+    }
+
+    public void NullCoalescenceAssignmentResult_NotNull()
+    {
+        int? i = null;
+        i ??= 1;
+        var r1 = (int)i;
+    }
+}
+
+public class SwitchExpressions
+{
+    public void Nullable_In_Arm_Noncompliant()
+    {
+        int? i = null;
+        int? result = i switch
+        {
+            null => (int) i, // FIXME Non-compliant
+            _ => 0
+        };
+    }
```

Comment:
It would be better for `i` to be a parameter. Otherwise, it is unclear whether this is non-compliant due to the null assignment or the null pattern.
```suggestion
    public void Nullable_In_Arm_Noncompliant(int? i)
    {
        int? result = i switch
        {
            null => (int) i, // FIXME Non-compliant
            _ => 0
        };
    }
```

Reply:
I'll apply this in #7307

Reply:
I've added both scenarios, as each has a reason to raise